compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -469,17 +469,15 @@ impl<'a> Linker for GccLinker<'a> {
         // eliminate the metadata. If we're building an executable, however,
         // --gc-sections drops the size of hello world from 1.8MB to 597K, a 67%
         // reduction.
-        } else if !keep_metadata {
+        } else if self.sess.target.linker_is_gnu && !keep_metadata {
```

Comment:
Does this also exclude lld?

Reply:
No, it should be a drop-in replacement. So if GNU ld accepts an option but `ld.lld` doesn't then that's a bug on lld's side.

Reply:
wasm32-unknown-unknown and wasm32-wasi don't set linker_is_gnu, but do use lld, which supports `--gc-sections`.

Reply:
Well, I guess more accurately lld tries to a drop-in replacement for the system linker on each platform. You can sorta see that if you try to just run lld verbatim:
```console
$ lld
lld is a generic driver.
Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead
```

So for wasm it uses a different frontend and for rustc, that means a different impl of the `Linker` trait here:
https://github.com/rust-lang/rust/blob/3fe1d7f78913231cdb7a8748149f7b2e67b814bd/compiler/rustc_codegen_ssa/src/back/linker.rs#L1219-L1221